### DISCUSSION TITLE
How do I set an AlpineJS model to an object returned by a function call?

### DISCUSSION THREAD

**Discussion by EricBusch** [alpinejs/alpine#648] [4. General]
Hi,

I'm trying to set the `this.results` model to an object returned by a call to [`fuse.search()`](https://fusejs.io/demo.html) but no matter what I try, the `this.results` is not getting set to the object returned by the `search()` function.

However `fuse.search()` does return the object I want in the console.

**Codepen:** https://codepen.io/LY7PV/pen/YzwvYqM?editors=1111

What am I doing wrong?

Thanks!

**Comment by KevinBatdorf** [comment]
You have to wait for the results.

```js
search: async function () {
  if (!this.query.length) {
    this.results = [{title: 'Nada'}]
    return
  }
  this.results = await fuse.search(this.query)
}
```

**Reply by EricBusch** [reply]
Thanks for your help! For some reason I wasn't accessing the `results` object correctly. I had to do the following instead. Still not sure why:

```html
<template x-for="(result, index) in results" :key="index">
    <div x-text="result.item.title"></div>
</template>
```

**Reply by SimoTod** [reply]
@EricBusch Your first example would have worked if you had used `item.item.title`. 
Results is the array you get back from fuse.
In your x-for you looped on each item an assigned them to a variable called item (this is a different item, not the one you get back from fuse, you probably got confused because you used the same name).
Each item is something like
```
{
    "item": {
      "title": "Old Man's War",
       ... 
    },
    "refIndex": 0,
    "score": 2.220446049250313e-16
  }
```
So you need to access the item property, then title.

**Reply by KevinBatdorf** [reply]
I thought maybe it wouldn't update if Fuse takes it's time to reply but I guess it does :)
You can also update it to `<template x-for="({item}, index) in results" :key="index">` but you won't have IE support

**Reply by EricBusch** [reply]
@SimoTod Thanks, I finally figured that out. I was using [this Vue code](https://github.com/tightenco/jigsaw-blog-template/blob/master/source/_assets/js/components/Search.vue) as an example so I wasn't expecting to have to use the extra `item` property in the loop.

@KevinBatdorf Regarding `x-for="({item}, index) in results"`, that's a cool tip! Thanks!
